### ISSUE TITLE
fix: jaspr build shutting down after running the command once, also same for jaspr serve

### ISSUE THREAD

**Issue by its-me-mahmud** [schultek/jaspr#322]
**Description**

After running the `jaspr build` command once, subsequent attempts result in an error where the build fails due to socket binding issues. The error suggests that the server is unable to bind to port 8080 due to an "address already in use" error. This issue requires manually ending all dart.exe processes via Task Manager in **Windows 10** for the command to work again.

<img src="https://github.com/user-attachments/assets/56ce7099-39ba-44bb-b35b-9f2899b02b53" height="500">

Error `jaspr build`:

```sh
$ jaspr build
Building jaspr for static rendering mode.
✓ Completed building web assets. (206.7s)
✗ [SERVER] [ERROR] Unhandled exception: (13.0s)
[SERVER] [ERROR] SocketException: Failed to create server socket (OS Error: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[SERVER] [ERROR] , errno = 10048), address = 0.0.0.0, port = 8080
[SERVER] [ERROR] #0      _NativeSocket.bind (dart:io-patch/socket_patch.dart:1032:7)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #1      _RawServerSocket.bind.<anonymous closure> (dart:io-patch/socket_patch.dart:1818:15)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #2      _ServerSocket.bind.<anonymous closure> (dart:io-patch/socket_patch.dart:2103:15)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #3      _HttpServer.bind.<anonymous closure> (dart:_http/http_impl.dart:3190:27)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #4      serve (package:shelf/shelf_io.dart:62:16)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #5      ServerApp._createServer (package:jaspr/src/server/server_app.dart:63:21)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #6      ServerApp._run (package:jaspr/src/server/server_app.dart:33:28)
[SERVER] [ERROR] <asynchronous suspension>
✗ [SERVER]  (10.0s)
Shutting down...
```

Error `jaspr serve`:

```sh
$ jaspr serve
Running jaspr in static rendering mode.
✓ Done building web assets. (45.1s)
✓ Server started. (22ms)
[SERVER] The Dart VM service is listening on http://127.0.0.1:8181/NMzb3a2uAp4=/
[SERVER] The Dart DevTools debugger and profiler is available at: http://127.0.0.1:8181/NMzb3a2uAp4=/devtools/?uri=ws://127.0.0.1:8181/NMzb3a2uAp4=/ws
[SERVER] [INFO] Server hot reload is enabled.
[SERVER] [ERROR] Unhandled exception:
[SERVER] [ERROR] SocketException: Failed to create server socket (OS Error: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[SERVER] [ERROR] , errno = 10048), address = 0.0.0.0, port = 8080
[SERVER] [ERROR] #0      _NativeSocket.bind (dart:io-patch/socket_patch.dart:1032:7)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #1      _RawServerSocket.bind.<anonymous closure> (dart:io-patch/socket_patch.dart:1818:15)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #2      _ServerSocket.bind.<anonymous closure> (dart:io-patch/socket_patch.dart:2103:15)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #3      _HttpServer.bind.<anonymous closure> (dart:_http/http_impl.dart:3190:27)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #4      serve (package:shelf/shelf_io.dart:62:16)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #5      ServerApp._createServer (package:jaspr/src/server/server_app.dart:63:21)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] [ERROR] #6      ServerApp._run (package:jaspr/src/server/server_app.dart:33:28)
[SERVER] [ERROR] <asynchronous suspension>
[SERVER] 
Shutting down...
Terminating web compiler...
```

**Steps To Reproduce**

1. Run the `jaspr build` command.
2. After the build completes, run either the `jaspr build` or `jaspr serve` command again.
3. The build fails with the error:
```sh
SocketException: Failed to create server socket (OS Error: Only one usage of each socket address (protocol/network address/port) is normally permitted.)
```
4. Open **Task Manager** in **Windows 10** and observe multiple `dart.exe` processes running.
5. End all `dart.exe` tasks in **Task Manager**.
6. Run the `jaspr build` command again, which now works successfully. 

**Doctor Output**

```sh
$ jaspr doctor
[✓] Jaspr CLI (Version 0.15.1)
  • Dart Version 3.5.3 (stable) (Wed Sep 11 16:22:47 2024 +0000) on "windows_x64" at C:\Users\abdullah\flutter\bin\cache\dart-sdk\bin\dart.exe   
  • Running on windows "Windows 10 Home" 10.0 (Build 19045) - Locale en-US
  • Analytics: Enabled

[✓] Current Project
  • Dependencies on core packages:
    • jaspr: ^0.15.1
    • jaspr_builder: ^0.15.1 (dev)
    • jaspr_web_compilers: ^4.0.10 (dev)
    • jaspr_test: ^0.15.1 (dev)
  • Rendering mode: static
  • Uses jaspr compilers: true
  • Uses flutter embedding: false
```

**Expected Behavior**

The `jaspr build` and `jaspr serve` command should complete successfully without requiring manual intervention to kill `dart.exe` processes between builds.

**Additional Context**

- The error occurs due to port 8080 being locked by previous instances of `dart.exe` that were not properly terminated.
- A screenshot of the multiple `dart.exe` processes in **Task Manager** is attached for reference:
```sh
SocketException: Failed to create server socket (OS Error: Only one usage of each socket address (protocol/network address/port) is normally permitted.)
```

**Comment by exoad** [comment]
I have had the same issue on release `0.16.0` and most of the time I have to stop the `dartaotruntime.exe` resulting in 

```
(OS Error: Only one usage of each socket address (protocol/network address/port) is normally permitted.
```

**Comment by schultek** [comment]
Can you try modifying the cli code and testing, since I don't have a windows machine.

Setup the local dev for jaspr as in the [contribution guide](https://github.com/schultek/jaspr/blob/main/CONTRIBUTING.md).
Then set [this line](https://github.com/schultek/jaspr/blob/ea8b15fdfcb681d8711ce809c923d5a7519770b8/packages/jaspr_cli/lib/src/commands/build_command.dart#L177) to false. 
Run `melos run activate:cli`.

See if that fixes the error.

**Comment by schultek** [comment]
Should be fixed in `v0.16.1`
